src/modules/mount/main.py

```diff
@@ -81,29 +81,27 @@ def mount_partition(root_mount_point, partition, partitions):
     # If a swapfile is used, we also create a subvolume "@swap".
     # Finally we remount all of the above on the correct paths.
     if fstype == "btrfs" and partition["mountPoint"] == '/':
-        has_home_mount_point = False
+        # Root has been mounted to btrfs volume -> create subvolumes from configuration
+        btrfs_subvolumes = libcalamares.job.configuration.get("btrfsSubvolumes") or []
+        subvolume_mountpoints = [d['mountPoint'] for d in btrfs_subvolumes]
+        # Check if listed mountpoints besides / are already present and don't create subvolume for those
         for p in partitions:
             if "mountPoint" not in p or not p["mountPoint"]:
                 continue
-            if p["mountPoint"] == "/home":
-                has_home_mount_point = True
-                break
-        needs_swap_subvolume = False
+            if p["mountPoint"] in subvolume_mountpoints and p["mountPoint"] != '/':
+                btrfs_subvolumes = [d for d in btrfs_subvolumes if d['mountPoint'] != p["mountpoint"]]
+        # Check if we need a subvolume for swap file
         swap_choice = global_storage.value( "partitionChoices" )
         if swap_choice:
             swap_choice = swap_choice.get( "swap", None )
             if swap_choice and swap_choice == "file":
-                needs_swap_subvolume = True
-
-        subprocess.check_call(['btrfs', 'subvolume', 'create',
-                               root_mount_point + '/@'])
-
-        if not has_home_mount_point:
+                btrfs_subvolumes.append({'mountPoint': '/swap', 'subvolume': '/@swap'})
+        # Store created list in global storage so it can be used in the fstab module
+        libcalamares.globalstorage.insert("btrfsSubvolumes", btrfs_subvolumes)
```

Comment:
Maybe log here what is being stored, so you're sure from the log that it was set?

Reply:
Good idea,I was testing it with added print statements in local branch.